src/pool-bin/libraries/BinHelper.sol

```diff
@@ -148,31 +150,41 @@ library BinHelper {
     /// @dev Returns the composition fees when adding liquidity to the active bin with a different
     ///      composition factor than the bin's one, as it does an implicit swap
     /// @param binReserves The reserves of the bin
-    /// @param fee 100 = 0.01%, 1000 = 0.1%
+    /// @param protocolFee 100 = 0.01%, 1000 = 0.1%
+    /// @param lpFee 100 = 0.01%, 1000 = 0.1%
     /// @param amountsIn The amounts of tokens to add
     /// @param totalSupply The total supply of the liquidity book
     /// @param shares The share of the liquidity book that the user will receive
     /// @return fees The encoded fees that will be charged
     function getCompositionFees(
         bytes32 binReserves,
-        uint24 fee, // fee: 100 = 0.01%
+        uint24 protocolFee, // fee: 100 = 0.01%
+        uint24 lpFee,
```

Comment:
Reason why i pass-in both fee and calculate swap fee inside:

Because protocolFee includes two direction i.e. 0to1 and 1to0. we'll need more gas to figure out which way it's going if we do it outside